PyRDF/backend/Utils.py

```diff
@@ -1,10 +1,41 @@
-import ROOT
-import os
 import logging
+import multiprocessing
+import os
+import threading
+
+try:
+    import queue
+except ImportError:
+    import Queue as queue
+
+import ROOT
 
 logger = logging.getLogger(__name__)
 
 
+def run_graphs(proxies, numthreads=multiprocessing.cpu_count()):
```

Comment:
Does this `numthreads` parameter really determines the number of event loops that are going to run in parallel?

Reply:
Correct, the function spawns exactly `numthreads` python `threading.Thread` objects. Each thread will submit a Spark job separately, which in turn will run a distributed event loop on the cluster. So `numthreads` is also the number of event loops that will be running distributedly and concurrently on the cluster at the same time.

The main reason is to avoid to spawn too many python threads at the same time, for example in an analysis with more samples than available threads on the driver

Reply:
Yes, I agree it's good to have it as limiting factor. What I'm not so sure about is the default, since a priori the number of cores in my client machine is not necessarily related to the number of concurrent jobs I am able to submit to a particular Spark cluster.

Reply:
Fair point, I just put that number as a first approximation. I think it should be somewhere between 4-8, anyway it's a parameter so a user could also ask for more threads to submit more concurrent jobs

Reply:
RunGraphs defaults to 4 concurrent job submissions for the moment